angr_platforms/risc_v/arch_riscv.py

```diff
@@ -1,60 +1,171 @@
+try:
+    import capstone as _capstone
+except ImportError:
+    _capstone = None
+
+from .arch import Arch, register_arch, Endness, Register
+from .archerror import ArchError
+from .tls import TLSArchInfo
+
 from archinfo.arch import register_arch, Arch, Register
 from archinfo.tls import TLSArchInfo
 
-#copied from arch msp430
+# copied from arch msp430
+
+
 class ArchRISCV(Arch):
-    def __init__(self, endness="Iend_LE"):
+
+    def __init__(self, endness=Endness.LE):
+        if endness != Endness.LE:
+            raise ArchError('Arch RISCV must be little endian')
         super(ArchRISCV, self).__init__(endness)
         self.call_pushes_ret = False
         self.branch_delay_slot = False
+
     sizeof = {'short': 16, 'int': 32, 'long': 32, 'long long': 64}
-    function_prologs = {}
-    function_epilogs = {}
 
     bits = 32
-    name = "RISCV"
-    instruction_endness = "Iend_LE" 
+    name = 'RISCV'
+
+    vex_arch = None  # No VEX support
+    qemu_name = None  # No Unicorn-engine support
+    ida_processor = None
+    triplet = 'riscv32-linux-gnu'
+    max_inst_bytes = 4
+
+    ip_offset = 128  # ip(pc)
+    sp_offset = 8  # sp(x2)
+    bp_offset = 8  # bp(x2)
+    lr_offset = 4  # lr(x1)
+    ret_offset = 40  # a0(x10) For return value
+
+    syscall_num_offset = 68
+    # a7(x17) For syscall number
+    # According To
+    # http://www.cs.uwm.edu/classes/cs315/Bacon/Lecture/HTML/ch05s03.html
+    call_pushes_ret = False
+    stack_change = -4
+
+    memory_endness = Endness.LE
+    register_endness = Endness.LE
+    instruction_endness = Endness.LE
+    sizeof = {'short': 16, 'int': 32, 'long': 32, 'long long': 64}
+
+    if _capstone:
+        try:
+            cs_arch = _capstone.CS_ARCH_RISCV
+            cs_mode = _capstone.CS_MODE_RISCV32
+        except AttributeError:
+            raise ArchError(
```

Comment:
The previous way you did this was better - this logic will make you unable to import this module if capstone is installed but not 5.0